webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonResponseObjects.scala

```diff
@@ -1914,7 +1914,111 @@ object JsonResponseObjects {
   }
 
   /**
-   * Representation of a group category with bare minimum group information
+   * Data container for the whole group category tree, provides the "groupCategories" dataContainer
+   */
+  final case class JRGroupCategoriesFull(groupCategories: JRFullGroupCategory)
+
+  /**
+   * Data container for the group category tree with minimal info, provides the "groupCategories" dataContainer
+   */
+  final case class JRGroupCategoriesMinimal(groupCategories: JRMinimalGroupCategory)
+
+  /**
+   * Representation of a group category with full group information
+   */
+  final case class JRFullGroupCategory(
+      id:                                     NodeGroupCategoryId,
+      name:                                   String,
+      description:                            String,
+      parent:                                 NodeGroupCategoryId,
+      @jsonField("categories") subCategories: List[JRFullGroupCategory],
+      groups:                                 List[JRGroup],
+      @jsonField("targets") targetInfos:      List[JRRuleTargetInfo]
+  )
+
+  object JRFullGroupCategory {
+    /*
+     * Somehow the chimney transformer derivation does not provide a way to pass the parent due to recursive type,
+     * so we have to make an inline transformer.
```

Comment:
I could not figure a solution to make a `implicit lazy val transformer: Transformer[JQGroupCategory, FullNodeGroupCategory]` declaration otherwise than inline (as in Rules), the culprit is having the `parent` being passed down so the parent value is relative but it is not an attribute of the `JQGroupCategory`